src/main/java/org/apache/sysds/hops/AggBinaryOp.java

```diff
@@ -306,9 +314,21 @@ protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
 				sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
 		}
 		*/
-		//currently always estimated as dense in order to account for dense intermediate without unnecessary overestimation 
-		double ret = OptimizerUtils.estimateSizeExactSparsity(dim1, dim2, sparsity);
-		
+		Hop input1 = getInput().get(0);
+		if(isMatrixMultiply() && hasCompressedInput() && !isRequiredDecompression() && input1.isCompressedOutput()) {
+
+			// right matrix multiplication ... compressed output
+			setCompressedOutput(true);
+			// conservatively set the size to a multiplication of the compression size
+			// multiplied with the number of columns.
+			setCompressedSize(input1._compressedSize * dim2);
+			ret = getCompressedSize();
```

Comment:
I would recommend moving the propagation of these properties into refreshSizeInformation()